src/posix/platform/include/openthread/openthread-system.h

```diff
@@ -311,6 +312,24 @@ bool otSysInfraIfIsRunning(void);
  */
 void otSysCliInitUsingDaemon(otInstance *aInstance);
 
+/**
+ * Sets whether to retrieve upstream DNS servers from "resolv.conf".
+ *
+ * @param[in] aEnabled  TRUE if enable retrieving upstream DNS servers from "resolv.conf", FALSE otherwise.
+ *
+ */
+void otSysUpstreamDnsServerSetResolvConfEnabled(bool aEnabled);
```

Comment:
What is the use case for making this a run-time configuration?

Reply:
On Android platform there isn't a system level `resolv.conf` so we need to have this configuration.

Regarding build-time vs runtime configuration, I think the runtime configuration's implementation is easier to maintain because the GitHub CI can ensure it's buildable. If we use a build time configuration, we'll have to guard some members by macro, which is likely not verified by GitHub CI. My preference is not strong though.